### PR TITLE
Rsg 11

### PR DESCRIPTION
**Pull request by Ivanm21**

### CODE REVIEW

File: src/Domain/Entities/ApplicationUser.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        public override string Id { get; set; }
+        public Gender Gender { get; set; }
+        public string Name { get; set; }
+        public string LastName { get; set; }
+        public string MidleName { get; set; }
+        public DateTime BirthDate { get; set; }
+
+        public string Passport { get; set; }
+        public int BSN { get; set; }
```
Comment (GasperPaul) [inline src/Domain/Entities/ApplicationUser.cs:20]: Shouldn't this be string?

File: src/Domain/Entities/ApplicationUser.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        public string Phone { get; set; }
+        
+        public List<ApplicationUser_UserGourp> ApplicationUser_UserGourps { get; set; }
+        public List<Customer> Customer { get; set; }
+
+    }
+
+    public enum Gender
+    {
+       male, female,other,none,unknown
```
Comment (GasperPaul) [inline src/Domain/Entities/ApplicationUser.cs]: Enum options should be PascalCase. Or in IMSCREAMINGIMFROMJAVACASE.

File: src/Domain/Entities/Customer.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Domain.Entities
+{
+    public class Customer
```
Comment (GasperPaul) [inline src/Domain/Entities/Customer.cs:8]: They are not customers.

File: src/Domain/Entities/Customer.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        public int CustomerId { get; set; }
+        public DateTime BirthDate { get; set; }
+        public DateTime Birthplace { get; set; }
+        public int Contact { get; set; }
```
Comment (GasperPaul) [inline src/Domain/Entities/Customer.cs:13]: Waht's this?

File: src/Domain/Entities/Customer.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        public int isActive { get; set; }
```
Comment (GasperPaul) [inline src/Domain/Entities/Customer.cs]: How this is int and not bool?

File: src/Domain/Entities/Customer.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        public int Resident { get; set; }
+        public DateTime ActivationDate { get; set; }
+        public DateTime DisactivationDate { get; set; }
+        public int RegisteredBy { get; set; }
+        public string SchoolGrade { get; set; }
```
Comment (GasperPaul) [inline src/Domain/Entities/Customer.cs:19]: M-m-m.. what?

File: src/DAL/ApplicationDbContext.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.Data.Entity;
+using Domain.Entities;
+using Microsoft.Data.Entity.Infrastructure;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace DAL
+{
+    public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
+    {
+        protected override void OnModelCreating(ModelBuilder builder)
```
Comment (GasperPaul) [inline src/DAL/ApplicationDbContext.cs:15]: You can remove unnecessary columns from Identity tables here.

File: src/WebUI/Startup.cs [hunk 8]
```diff
@@ -98,6 +103,11 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                     name: "default",
                     template: "{controller=Home}/{action=Index}/{id?}");
             });
+
+            //Seed DataBase
+            await dataInitializer.InitializeDataAsync();
```
Comment (GasperPaul) [inline src/WebUI/Startup.cs:108]: Wouldn't this run every app restart? Why not just put seeding in seed method called on migration as always?

**Comment by GasperPaul** [comment]
Guys, let's give a descriptive names to each commit, k tnx.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

```
 public string SchoolGrade { get; set; }
```

M-m-m.. what?

Also one of the fields of Customer entity from diagram
-            await dataInitializer.InitializeDataAsync();
  Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

**Comment by Ivanm21** [comment]
+   public class Customer
"They are not customers."
It is name of one of the Tables from provided diagram. I can either delete or rename it if needed,

+   public int Contact { get; set; }
It is one of the fields of Customer entity from provided diagram. I suppose that the purpose of the field is to distinguish if Customer is contactable or not. 

+  public int isActive { get; set; }
How this is int and not bool?

Sorry, i will update field shortly

+     public string SchoolGrade { get; set; }
M-m-m.. what?

Also one of the fields of Customer entity from diagram

+            await dataInitializer.InitializeDataAsync();
Yes, It will run every app restart. Unfortunatly Seed method is not yet implemented in EF7.

File: src/DAL/DataInitializer.cs [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+        public async Task InitializeDataAsync()
+        {
+           
+            await CreatRoles();
+            await CreateUsersAsync();
+            await CreateUserGroup();
+        }
+
+        private async Task CreatRoles()
```
Comment (askliar) [inline src/DAL/DataInitializer.cs:32]: @ivan_m21 how does this method work?

Comment (askliar) [inline src/DAL/DataInitializer.cs:32]: @ivanm21

**Comment by Ivanm21** [comment]
@emohat1r It runs at the application start and seeds db if it is not yet seeded.

**Comment by Ivanm21** [comment]
@emohat1r It runs at the application start and seeds db if it was not yet seeded.

**Comment by GasperPaul** [comment]
@Ivanm21

> ``` csharp
> public class Customer
> ```

We still  need to get proper naming conventions and nomenclature. @emohat1r, any luck with that?

> ``` csharp
> public int Contact { get; set; }
> ...
> public string SchoolGrade { get; set; }
> ```

@emohat1r, we need them to explain. It's probably something we already though of. 
School grade, really? How they think that should be?

> Unfortunatly Seed method is not yet implemented in EF7.

Yep, that's sad. My bad for pointing that out. It's totaly ok architecture then.

**Comment by askliar** [comment]
@GasperPaul Yes, I will talk with them tomorrow and fully discuss each of our questions, so that we have all workflows as soon as possible

**Pull request merged**